### PR TITLE
Merge ign-fuel-tools7 ➡️  gz-fuel-tools8

### PR DESCRIPTION
**Pull request by scpeters**
# ➡️  Forward port

Port `ign-fuel-tools7 ` ➡️  `gz-fuel-tools8`

Branch comparison: https://github.com/gazebosim/gz-fuel-tools/compare/gz-fuel-tools8...ign-fuel-tools7

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

**Comment by scpeters** [comment]
@azeey I resolved some conflicts from https://github.com/gazebosim/gz-fuel-tools/commit/fe66f33b3235a70855a43e79437653612cea1897; hopefully I got it right

File: src/FuelClient_TEST.cc [hunk 1]
```diff
@@ -1366,6 +1366,52 @@ TEST_F(FuelClientTest, Models)
   }
 }
 
+//////////////////////////////////////////////////
+TEST_F(FuelClientTest, ModelsCheckCached)
+{
+  ClientConfig config;
+  std::string cacheDir = common::joinPaths(PROJECT_BINARY_PATH, "test_cache");
```
Comment (scpeters) [inline src/FuelClient_TEST.cc:1373]: this needs to be fixed since `PROJECT_BINARY_PATH` is no longer in use

**Review by azeey — APPROVED** [review]

File: src/FuelClient_TEST.cc (hunk 1 above)
Comment (scpeters) [reply src/FuelClient_TEST.cc]: good call, I just amended to https://github.com/gazebosim/gz-fuel-tools/pull/362/commits/cfa86f7eaf84eb5b63a334651a4a8d90bb710989

**Comment by scpeters** [comment]
the windows test failures are already present on the `gz-fuel-tools8` branch:

* https://build.osrfoundation.org/view/ign-garden/job/ign_fuel-tools-gz-8-win/60/

**Pull request merged**